### PR TITLE
Port training and evaluation code

### PR DESCRIPTION
**Pull request by drubinstein**
Beginning the work on bringing over the training and evaluation code. Please point out if I have to remove any keywords that exist.

I have not tested this code. This code will also require a lot of documentation and style adjustments e.g. typing and testing + possibly some generalization beyond GCP. The version of mirdata used may also be slightly out of date and there probably needs to be a better system for downloading the mir datasets.

Theoretically you should be able run the individual pipelines in dataset.py with DirectRunner then run the training script and subsequent evaluation.

### CODE REVIEW

**Comment by joakimunge** [comment]
@drubinstein Curious about progress on this. Is this usable on some capacity? I'm looking into fine tuning the model for a pet project

**Comment by migperfer** [comment]
Hi! @drubinstein many thanks for opening this PR. I wanted to know more about the splits used for GuitarSet, and I noticed that `basic_pitch/experiments/split_ids/guitarset_test_tracks.csv` contains 30 files, while [the paper](https://arxiv.org/pdf/2203.09893.pdf) mentions that the test set is composed of 72 files. Is the `.csv` uploaded for a split based on a later iteration of the model's experiments?

**Comment by matangover** [comment]
I'm wondering about something in the code: Why was `tf_example_deserialization.get_sample_weights` designed to ignore empty targets for the purpose of loss calculation? I can't understand the reasoning. In most circumstances if the `notes` target is empty, so would `contour` and `notes` be. Why not ignore these examples completely? The training would be faster since it wouldn't have to go through those "empty". Or, otherwise, why not include the empty examples in the training?

**Comment by matangover** [comment]
> Curious about progress on this. Is this usable on some capacity? I'm looking into fine tuning the model for a pet project

@joakimunge Yes, I've successfully fine tuned it

**Comment by matangover** [comment]
The paper mentions:
> During training, random label-preserving augmentations are applied to the audio, including adding noise, equalization filters, and reverb.

Would you be able to share this data augmentation code? I don't see it in this PR

**Comment by drubinstein** [comment]
Closing with preference given to #99

**Comment by Vosatorp** [comment]
Why training code is not merged?

**Comment by drubinstein** [comment]
> Why training code is not merged?
We only have so much developer free time to put into merging training code and abstract it away from our internal infrastructure. Currently, we're almost done merging in one dataset for the data preprocessing code (#99). After that we need to merge in the rest of the dataset preprocessing and then we can get to training.

> Why basic pitch written on tensorflow, not torch?
Long term we plan to port it to torch, but Basic Pitch was written when we were using tensorflow internally.
